### PR TITLE
New version: py-qtconsole 5.2.0

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-qtconsole/package.py [hunk 1]
```diff
@@ -12,17 +12,21 @@ class PyQtconsole(PythonPackage):
...
     depends_on('py-ipykernel@4.1:',      type=('build', 'run'))
     depends_on('py-jupyter-client@4.1:', type=('build', 'run'))
     depends_on('py-jupyter-core',        type=('build', 'run'))
     depends_on('py-pygments',            type=('build', 'run'))
     depends_on('py-traitlets',           type=('build', 'run'))
     depends_on('py-ipython-genutils',    type=('build', 'run'), when='@4.5.1:')
     depends_on('py-sphinx@1.3:',         type=('build', 'run'), when='+docs')
-    depends_on('py-pyqt5',               type='run')
+    depends_on('py-pyqt5',               type='run', when='@:5.1')
+    depends_on('py-qtpy',                type='run', when='@5.2.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-qtconsole/package.py]:
```suggestion
    depends_on('py-qtpy',                type=('build', 'run'), when='@5.2.0:')
```

File: var/spack/repos/builtin/packages/py-qtconsole/package.py [hunk 2]
```diff
@@ -12,17 +12,21 @@ class PyQtconsole(PythonPackage):
...
+    depends_on('py-pyzmq@17.1:',         type='run', when='@5.2.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-qtconsole/package.py]:
```suggestion
    depends_on('py-pyzmq@17.1:',         type=('build', 'run'), when='@5.2.0:')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by iarspider** [comment]
`py-qtpy` and `py-pyzmq` are not, technically speaking, required at build-time (and, as far as I can tell, are not downloaded/installed), I still agree that for consistency these should be `type=('build', 'run')`.

**Comment by adamjstewart** [comment]
> `py-qtpy` and `py-pyzmq` are not, technically speaking, required at build-time (and, as far as I can tell, are not downloaded/installed), I still agree that for consistency these should be `type=('build', 'run')`.

If you already happen to have these installed and present in your `PYTHONPATH`, they won't be downloaded, but if they aren't, then setuptools/pip will, which breaks air-gapped systems.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**